src/main/scala-2.12/com/linagora/gatling/imap/scenario/ImapCompressScenario.scala

```diff
@@ -0,0 +1,15 @@
+package com.linagora.gatling.imap.scenario
+
+import com.linagora.gatling.imap.PreDef._
+import io.gatling.core.Predef._
+import io.gatling.core.feeder.FeederBuilder
+import io.gatling.core.structure.ScenarioBuilder
+
+object ImapCompressScenario {
+  def apply(feeder: FeederBuilder): ScenarioBuilder =
+    scenario("Imap")
+      .feed(feeder)
+      .exec(imap("Connect").connect()).exitHereIfFailed
+      .exec(imap("capability").capability().check(contains("COMPRESS=DEFLATE")))
+      .exec(imap("compress").compress("deflate").check(ok))
```

Comment:
Please add a noop command after to ensure the connection is still usable once compression is turned on.